src/seedu/addressbook/data/person/Name.java

```diff
@@ -77,6 +77,19 @@ public boolean isSimilar(Name other) {
         if (lowerThisName.equals(lowerOtherName)){
             return true;
         }
+        
+        List<String> wordsInThis = this.getWordsInName();
+        List<String> wordsInOther = ((Name)other).getWordsInName();
+        
+        //Finds any common words between the words this Name and the words in the other Name
+        for (String thisWord : wordsInThis){
+            for(String otherWord : wordsInOther){
+                if(thisWord.toLowerCase().equals(otherWord.toLowerCase())){
+                    return true;
```

Comment:
If I remember correctly, this doesn't work properly because once this condition is satisfied it will return true and exit the method.